File: backend/app/schemas/match.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class JobInfo(BaseModel):
    external_id: str
    title: str
    company: str
    description: str
    location: Optional[str] = None
    salary_min: Optional[float] = None
    salary_max: Optional[float] = None
    job_type: Optional[str] = None
    source: Optional[str] = None
    apply_url: str
    deadline: Optional[datetime] = None
    
    class Config:
        from_attributes = True

class MatchBase(BaseModel):
    user_id: int
    cv_id: int
    external_job_id: str

class MatchResponse(MatchBase):
    id: int
    match_score: float
    match_reasons: Optional[str] = None
    status: str
    created_at: datetime
    updated_at: Optional[datetime] = None
    job: Optional[JobInfo] = None
    
    class Config:
        from_attributes = True

class MatchUpdate(BaseModel):
    status: str

File: frontend/src/types/Match.ts
import { Job } from './Job';

export interface Match {
  id: number;
  user_id: number;
  cv_id: number;
  external_job_id: string;
  match_score: number;
  match_reasons?: string;
  status: string;
  created_at: string;
  updated_at?: string;
  job?: Job;
}

export interface MatchUpdate {
  status: string;
}
